src/pages/Answers.jsx

```diff
@@ -24,80 +25,52 @@ export default function Answer({ user }) {
   const [errorMessage, setErrorMessage] = useState('Something went wrong. Check the console for more information.');
   const messageStore = useMessageStore();
 
-  useEffect(() => {
-    if (question_id) {
-      toggleLoading(true);
-      let axiosConfig = {
-        method: 'get',
-        url: `/api/questions/${question_id}`,
-        headers: {
-          'Content-Type': 'application/json',
-        },
-      };
-      if (user) {
-        axiosConfig.headers.authorization = `Bearer ${user.id_token}`;
-      }
-      axios.request(axiosConfig)
-        .then((res) => {
-          // console.log('question', res.data);
-          axiosConfig = {
-            method: 'get',
-            url: '/api/answers',
-            params: { question_id },
-            headers: {
-              'Content-Type': 'application/json',
-            },
-          };
-          if (user) {
-            axiosConfig.headers.authorization = `Bearer ${user.id_token}`;
-          }
-          axios.request(axiosConfig)
-            .then((response) => {
-              // console.log('answers', response.data);
-              if (response.data && Array.isArray(response.data)) {
-                axiosConfig = {
-                  method: 'get',
-                  url: `/api/answers/${response.data[0].id}`,
-                  headers: {
-                    'Content-Type': 'application/json',
-                  },
-                };
-                if (user) {
-                  axiosConfig.headers.authorization = `Bearer ${user.id_token}`;
-                }
-                axios.request(axiosConfig)
-                  .then((res3) => {
-                    // console.log('answer', res3.data);
-                    const { knowledge_graph, results } = res3.data;
-                    const query_graph = res.data;
-                    const message = { query_graph, knowledge_graph, results };
-                    messageStore.initializeMessage(message);
-                    setMessageSaved(true);
-                    toggleLoading(false);
-                  })
-                  .catch((err) => {
-                    console.log('err', err);
-                    setErrorMessage(`Unable to get answer: ${response.data[0].id}`);
-                    toggleLoading(false);
-                  });
-              } else {
-                setErrorMessage('This question has no visible answers.');
-                toggleLoading(false);
-              }
-            })
-            .catch((err) => {
-              console.log('err', err);
-              setErrorMessage(`Unable to retrieve answers for question: ${question_id}`);
-              toggleLoading(false);
-            });
-        })
-        .catch((err) => {
-          console.log('err', err);
-          setErrorMessage('Unable to load question.');
-          toggleLoading(false);
-        });
+  let token;
```

Comment:
Let's move this into the fetchQuestionAnswerData function, because this will otherwise be run on every rerender.